### PR TITLE
Getting Started Guide fixes

### LINKED ISSUE #3943: Getting Started Guide memo fields shouldn't strip out returns/new lines
Currently, when inputting a bunch of text into any of the memo/paragraph inputs in Edit mode of the Getting Started Guide, when you save your text, the Guide ignores any new lines or returns in the text. This makes long passages of text hard to read, and makes writing bulleted lists impossible.

What's interesting is that if you go back to edit mode, the new lines are still present in the saved text, so the ignoring is just happening at the presentation layer apparently.

### LINKED ISSUE #4635: Metrics not displaying in Getting Started Guide
In the Getting Started Guide, metrics that you include show up in the Edit state of the guide, but not in the actual published guide.

This is on Chrome on the master branch. Not seeing any server or JS errors.

### PR DESCRIPTION
**Pull request by tlrobinson**
* Fix metrics and misc other logic in GSG frontend. Resolves #4635
* Fix incorrect API validation
* Preserve whitespace in GSG content. Resolves #3943

### CODE REVIEW

File: src/metabase/api/dashboard.clj [hunk 1]
```diff
@@ -74,9 +74,9 @@
   [id :as {{:keys [description name parameters caveats points_of_interest show_in_getting_started enable_embedding embedding_params], :as dashboard} :body}]
   {name                    (s/maybe su/NonBlankString)
    description             (s/maybe su/NonBlankString)
-   caveats                 (s/maybe su/NonBlankString)
-   points_of_interest      (s/maybe su/NonBlankString)
-   show_in_getting_started (s/maybe su/NonBlankString)
+   caveats                 (s/maybe s/Str)
```
Comment (salsakran) [inline src/metabase/api/dashboard.clj]: Can we have a test for this?

File: frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx [hunk 2]
```diff
@@ -487,7 +487,7 @@ export default class ReferenceGettingStartedGuide extends Component {
                                     />
                                 </div>
                             ]}
-                            { Object.keys(metrics) > 0  && (
+                            { Object.keys(metrics).length > 0  && (
```
Comment (salsakran) [inline frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx]: Is there anyway to catch this sort of stuff?  Seems like a basic type error.

**Review by salsakran — CHANGES_REQUESTED** [review]

File: frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx (hunk 2 above)
Comment (tlrobinson) [inline frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx]: Flow would catch it. I've been adding annotations for most new code and as I come across things, but it's a fair amount of work.

File: frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx [hunk 3]
```diff
@@ -532,56 +532,53 @@ export default class ReferenceGettingStartedGuide extends Component {
 
                             <div className="mt4 pt4">
                                 <SectionHeader trim={(!has(guide.important_segments) && !has(guide.important_tables))}>
-                                    { Object.keys(segments) > 0 ? 'Segments and tables' : 'Tables' }
+                                    { Object.keys(segments).length > 0 ? 'Segments and tables' : 'Tables' }
```
Comment (kdoh) [inline frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx]: Can we use the same `has` logic check here?

Comment (tlrobinson) [inline frontend/src/metabase/reference/containers/ReferenceGettingStartedGuide.jsx]: `has` looks like it's different: `export const has = (entity) => entity && entity.length > 0;`

File: src/metabase/api/dashboard.clj (hunk 1 above)
Comment (tlrobinson) [inline src/metabase/api/dashboard.clj]: Done

**Review by kdoh — APPROVED** [review]

**Pull request merged**